### PR TITLE
Timetable validation viewer

### PR DESCRIPTION
**Pull request by evansiroky**
Show a preview of timetables in the feed viewer

### CODE REVIEW

File: lib/common/util/__tests__/gtfs.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+      it('should parse a value in the day', () => {
+        expect(secondsAfterMidnightToHHMM(12345)).toEqual('03:25:45')
+      })
+
+      it('should parse a value after midnight', () => {
+        expect(secondsAfterMidnightToHHMM(99999)).toEqual('27:46:39')
+      })
+
+      it('should parse a value 2 days in the future', () => {
```
Comment (landonreed) [inline lib/common/util/__tests__/gtfs.js:20]: Is this actually something we want to permit?  I suppose any source GTFS could contain a value that extends this long into the future, but perhaps we should log a warning or something.

File: lib/common/util/__tests__/gtfs.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        expect(secondsAfterMidnightToHHMM(222222)).toEqual('61:43:42')
+      })
+
+      it('should parse a value below 0', () => {
```
Comment (landonreed) [inline lib/common/util/__tests__/gtfs.js:24]: When do we need negative seconds parsed into past times?  Should negative values log an error?

File: lib/manager/components/reporter/containers/Timetables.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+const mapStateToProps = (state, ownProps) => {
+  const {
+    gtfs: {
+      filter,
+      patterns,
+      routes,
+      services,
+      timetables
+    }
+  } = state
```
Comment (landonreed) [inline lib/manager/components/reporter/containers/Timetables.js]: Typically when I destructure I use the syntax `const {filter} = state.gtfs`.  I find that a little easier to read.

File: lib/manager/components/reporter/containers/Timetables.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+const mapDispatchToProps = (dispatch, ownProps) => {
+  const {namespace} = ownProps.version
+  return {
+    onComponentMount: (initialProps) => {
+      if (!initialProps.routes.fetchStatus.fetched) {
+        dispatch(fetchRoutes(namespace))
+      }
+    },
+    routeFilterChange: (newValue) => {
```
Comment (landonreed) [inline lib/manager/components/reporter/containers/Timetables.js:43]: I thought you had expressed an interest in moving away from longform `mapDispatchToProps` and towards supplying a simple json object. I'm somewhat in favor of that style as well.  Perhaps this container should reflect that as well.

File: lib/manager/selectors/index.js [hunk 5]
```diff
@@ -1,11 +1,205 @@
...
+        )
+        row[`stop-${stopTime.stop_id}-departure-raw`] = stopTime.departure_time
+        row[`stop-${stopTime.stop_id}-departure`] = secondsAfterMidnightToHHMM(
+          stopTime.departure_time
+        )
+      })
+      return row
+    })
+
+    rows.sort((a, b) => {
```
Comment (landonreed) [inline lib/manager/selectors/index.js]: This could use a comment about the sorting logic.

File: lib/manager/selectors/index.js [hunk 6]
```diff
@@ -1,11 +1,205 @@
...
+        row: '0',
+        width: 150
+      }
+    ]
+
+    // make rest of the row headers by making stop names as headers
+    // assume that the first trip is representative of stops in timetable
+    const firstTrip = pattern.trips[0]
+
+    // sort stop times by stop_sequence just in case
```
Comment (landonreed) [inline lib/manager/selectors/index.js:125]: Does the sort really need to happen here?  Doesn't gtfs-lib handle that already?

File: lib/manager/selectors/index.js [hunk 7]
```diff
@@ -1,11 +1,205 @@
...
+    firstTrip.stop_times.sort((a, b) => a.stop_sequence - b.stop_sequence)
+    const stopColWidth = 200
+    firstTrip.stop_times.forEach(stopTime => {
```
Comment (landonreed) [inline lib/manager/selectors/index.js]: This could use comments about how the showArrivals filter is applied.

File: lib/manager/selectors/index.js [hunk 8]
```diff
@@ -1,11 +1,205 @@
...
+        tripShortName: trip.trip_short_name
+      }
+      trip.stop_times.forEach(stopTime => {
+        if (
+          isNumber(stopTime.arrival_time) ||
+          isNumber(stopTime.departure_time)
+        ) {
+          stopHasStopTimeWithTime[stopTime.stop_id] = true
+        }
+        row[`stop-${stopTime.stop_id}-arrival-raw`] = stopTime.arrival_time
```
Comment (landonreed) [inline lib/manager/selectors/index.js]: A comment about how these fields are used would be helpful (i.e., why is the raw value preserved?). Edit: ah, probably for sorting below.  Still, a comment would be good.

**Review by landonreed — CHANGES_REQUESTED** [review]
Looks mostly good.  Just a few notes about comments.

File: lib/common/util/__tests__/gtfs.js (hunk 1 above)
Comment (evansiroky) [reply lib/common/util/__tests__/gtfs.js:20]: Yes, this is an edge case.  As for logging a warning, that seems to be something that could be done in gtfs-lib.  This is merely a unit test for this little conversion function that covers all possible edge cases.

File: lib/common/util/__tests__/gtfs.js (hunk 2 above)
Comment (evansiroky) [reply lib/common/util/__tests__/gtfs.js:24]: Yes, this is an edge case.  As for logging a warning, that seems to be something that could be done in gtfs-lib.  This is merely a unit test for this little conversion function that covers all possible edge cases.

File: lib/manager/components/reporter/containers/Timetables.js (hunk 4 above)
Comment (evansiroky) [reply lib/manager/components/reporter/containers/Timetables.js:43]: I intend to work on that in one big refactor effort as outlined in https://github.com/catalogueglobal/datatools-ui/issues/182.  For now I'm just using whatever the existing coding style is.

File: lib/manager/components/reporter/containers/Timetables.js (hunk 3 above)
Comment (evansiroky) [reply lib/manager/components/reporter/containers/Timetables.js]: ok, when destructuring I'll only go down one level at time

File: lib/manager/selectors/index.js (hunk 6 above)
Comment (evansiroky) [reply lib/manager/selectors/index.js:125]: ah, I guess it does

**Comment by landonreed** [comment]
Closing because fixes for this were applied in #202.
